src/main/java/com/imsweb/datagenerator/provider/ProviderDataGenerator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2018 Information Management Services, Inc.
+ */
+package com.imsweb.datagenerator.provider;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.imsweb.datagenerator.DataGenerator;
+
+public abstract class ProviderDataGenerator implements DataGenerator {
+
+    // list of rules to be executed
+    protected List<ProviderDataGeneratorRule> _rules;
+
+    public ProviderDataGenerator() {
+        _rules = new ArrayList<>();
+    }
+
+    @Override
+    public abstract String getId();
+
+    /**
+     * Returns all the provider rules from this generator.
+     */
+    public List<ProviderDataGeneratorRule> getRules() {
+        return _rules;
+    }
+
+    /**
+     * Generates a single provider.
+     * <br/><br/>
+     * @param options options
+     * @return generated provider
+     */
+    public Map<String, String> generateProvider(ProviderDataGeneratorOptions options) {
+        // make sure options are never null
+        if (options == null)
+            options = new ProviderDataGeneratorOptions();
+
+        if ((options.getState() == null) || (options.getState().equals("")))
```

Comment:
You do not need the inner parenthesis, and it's usually better practice to call "String.isEmpty()" instead of comparing to the empty String. But in this case, there is a utility method you can use that checks null, blank, and blank-after-trimming, all in one:
```java
StringUtils.isBlank(options.getState)
```